### PR TITLE
Check physical phase space in B->3lnu observables

### LINKED ISSUE #729: B->l^- nu l'^+ l'^-: branching ratio can be evaluated outside the physical phase space
[First reported by Martino Borsato]

The comments and code in
https://github.com/eos/eos/blob/870d11b1b7cf07ab6ffbab708fbd2654ba2e6fff/eos/b-decays/b-to-3l-nu.cc#L91-L97
should be adjusted to respect the phase space boundary ``m_l^2 <= k2 <= (m_B - sqrt(q2))^2``.

### PR DESCRIPTION
**Pull request by SKuerten**
resolves #729

### CODE REVIEW

File: eos/b-decays/b-to-3l-nu.cc [hunk 1]
```diff
@@ -90,8 +90,8 @@ namespace eos
 
         /*!
          * differential decay width of 2 kinematic variables
-         * q2 is the invariant mass of the off-shell photon in the range 4 m_l'^2 <= q2 <= (m_B - k2)^2
-         * k2 is the invariant mass of the W-meson in the range m_l^2 <= k2 <= (m_B -q2)^2
+         * q2 is the invariant mass of the off-shell photon in the range 4 m_l'^2 <= q2 <= (m_B - sqrt(k2) )^2
```
Comment (dvandyk) [inline eos/b-decays/b-to-3l-nu.cc:93]: That does not reflect what we had written in [KKvDZ:2022A], which is: $4 m_{\ell'}^2 \leq q^2 \leq (m_B - m_\ell)^2$. $k^2$ seems to be correct.

File: eos/b-decays/b-to-3l-nu.cc [hunk 2]
```diff
@@ -683,6 +689,12 @@ namespace eos
             - 2 * k2 * (m_B_sq + q2)) * z_w) + k2 * (-m_B_sq + m_l_sq - q2 + sqrt(k4
             + m_B_sq_q2_diff2 - 2 * k2 * (m_B_sq + q2)) * z_w)));
 
+            if ( power_of<2>(m_B - sqrt(k2)) < q2 || q2 < 4 * m_lprime_sq
+                || power_of<2>(m_B - sqrt(q2)) < k2 || k2 < m_l_sq )
```
Comment (dvandyk) [inline eos/b-decays/b-to-3l-nu.cc:693]: See my earlier comment. This does not reflect what we have in the paper.

**Review by dvandyk — CHANGES_REQUESTED** [review]

**Comment by dvandyk** [comment]
Nitpicking on the commit messages. We typically do not use the file names in the commit messages, so ``B->3lnu`` rather than ``bto3lnu`` in the first commit message. Also, why did you split this into two commits? I think one would suffice.

**Comment by SKuerten** [comment]
> Nitpicking on the commit messages. We typically do not use the file names in the commit messages, so `B->3lnu` rather than `bto3lnu` in the first commit message. Also, why did you split this into two commits? I think one would suffice.

You are just to quick. I was already fixing that.^^

File: eos/b-decays/b-to-3l-nu.cc (hunk 1 above)
Comment (SKuerten) [reply eos/b-decays/b-to-3l-nu.cc]: fixed

File: eos/b-decays/b-to-3l-nu.cc (hunk 2 above)
Comment (SKuerten) [reply eos/b-decays/b-to-3l-nu.cc:693]: done

File: eos/b-decays/b-to-3l-nu.cc [hunk 3]
```diff
@@ -184,7 +184,13 @@ namespace eos
             + f25 * f_B * real(F_2) + f35 * f_B * real(F_3) + f45 * f_B * real(F_4)
             + f55 * power_of<2>(f_B);
 
-        return prefactor * amp;
```
Comment (dvandyk) [inline eos/b-decays/b-to-3l-nu.cc:187]: Thanks for fixing this, would have driven me mad(der) :smile:

**Review by dvandyk — APPROVED** [review]

**Pull request merged**